virtual_ecosystem/models/plants/canopy.py

```diff
@@ -201,3 +70,46 @@ def initialise_canopy_layers(data: Data, layer_structure: LayerStructure) -> Dat
     )
 
     return data
+
+
+def calculate_canopies(
+    communities: PlantCommunities, max_canopy_layers: int
+) -> dict[int, Canopy]:
+    """Calculate the canopy structure of communities.
+
+    This function takes a PlantCommunities object and calculates the canopy
+    representation for each community, using the perfect plasticity approximation to
+    calculate the closure heights of the canopy layers.
+
+    Args:
+        communities: The PlantCommunities object to convert
+        max_canopy_layers: The maximum number of permitted canopy layers
+
+    Returns:
+        A dictionary, keyed by cell id, of the canopy representations of each community.
```

Comment:
Since you import pyrealm here, it would be useful to give a bit more info on the actual outputs

Reply:
I've linked through to the `pyrealm` `Canopy` class, which gives the details docs on the returned values.